packages/e2e-test-app-fabric/windows/RNTesterApp-Fabric/RNTesterApp-Fabric.cpp

```diff
@@ -51,263 +46,200 @@ struct DeviceInfo {
-    if (m_CompositionHwndHost) {
-      return static_cast<LRESULT>(m_CompositionHwndHost.TranslateMessage(message, wparam, lparam));
-    }
-    return 0;
-  }
-};
+  return host;
+}
 
-extern "C" IMAGE_DOS_HEADER __ImageBase;
-HINSTANCE WindowData::s_instance = reinterpret_cast<HINSTANCE>(&__ImageBase);
+_Use_decl_annotations_ int CALLBACK WinMain(HINSTANCE instance, HINSTANCE, PSTR /* commandLine */, int showCmd) {
+  // Initialize WinRT.
+  winrt::init_apartment(winrt::apartment_type::single_threaded);
 
-LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam) {
-  auto windowData = WindowData::GetFromWindow(hWnd);
-  if (windowData) {
-    auto result = WindowData::GetFromWindow(hWnd)->TranslateMessage(message, wParam, lParam);
-    if (result)
-      return result;
-  }
+  // Enable per monitor DPI scaling
+  SetProcessDpiAwarenessContext(DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2);
 
-  switch (message) {
-    case WM_DESTROY: {
-      delete WindowData::GetFromWindow(hWnd);
-      SetProp(hWnd, WindowDataProperty, 0);
-      PostQuitMessage(0);
-      return 0;
-    }
-    case WM_NCCREATE: {
-      auto cs = reinterpret_cast<CREATESTRUCT *>(lParam);
-      auto windowData = static_cast<WindowData *>(cs->lpCreateParams);
-      WINRT_ASSERT(windowData);
-      SetProp(hWnd, WindowDataProperty, reinterpret_cast<HANDLE>(windowData));
-      break;
-    }
-    case WM_GETOBJECT: {
-      if (lParam == UiaRootObjectId) {
-        auto windowData = WindowData::GetFromWindow(hWnd);
-        if (windowData == nullptr || !windowData->m_windowInited)
-          break;
-
-        auto hwndHost = windowData->m_CompositionHwndHost;
-        winrt::com_ptr<IRawElementProviderSimple> spReps;
-        if (!hwndHost.UiaProvider().try_as(spReps)) {
-          break;
-        }
-        LRESULT lResult = UiaReturnRawElementProvider(hWnd, wParam, lParam, spReps.get());
-        return lResult;
-      }
-    }
-  }
+  // Create a DispatcherQueue for this thread.  This is needed for Composition, Content, and
+  // Input APIs.
+  auto dispatcherQueueController{winrt::Microsoft::UI::Dispatching::DispatcherQueueController::CreateOnCurrentThread()};
 
-  return DefWindowProc(hWnd, message, wParam, lParam);
-}
+  // Create a Compositor for all Content on this thread.
+  auto compositor{winrt::Microsoft::UI::Composition::Compositor()};
 
-int RunRNTester(int showCmd) {
-  auto windowData = std::make_unique<WindowData>(winrt::Microsoft::ReactNative::CompositionHwndHost());
-  HWND hwnd = CreateWindow(
-      c_windowClassName,
-      appName,
-      WS_OVERLAPPEDWINDOW,
-      CW_USEDEFAULT,
-      CW_USEDEFAULT,
-      CW_USEDEFAULT,
-      CW_USEDEFAULT,
-      nullptr,
-      nullptr,
-      WindowData::s_instance,
-      windowData.get());
-
-  WINRT_VERIFY(hwnd);
+  // Create a top-level window.
+  auto window = winrt::Microsoft::UI::Windowing::AppWindow::Create();
+  window.Title(appName);
```

Comment:
Should we expose a turbomodule so fabric apps can change their title from JS?

Reply:
That assumes there is one window.  -- And we know what it is, and some other things.  I think that would have to be part of a larger story to work out what windowing from JS looks like.